### PR TITLE
Update to ts5

### PR DESCRIPTION
**Pull request by mattgodbolt**

### CODE REVIEW

**Comment by mattgodbolt** [comment]
Looks like I'm hitting cypress-io/cypress#26203

File: package.json [hunk 1]
```diff
@@ -168,7 +168,7 @@
     "lint": "eslint --max-warnings=0 . --fix",
     "lint-check": "eslint --max-warnings=0 . && prettier --check .",
     "lint-files": "eslint --max-warnings=0",
-    "test": "node --no-warnings=ExperimentalWarning --loader ts-node/esm ./node_modules/mocha/bin/mocha.js -b 'test/**/*.ts' 'test/**/*.js'",
```
Comment (jeremy-rifkin) [inline package.json:171]: This will cause only the js tests to run

**Comment by jeremy-rifkin** [comment]
With ts5 are we able to get rid of the .js in imports from ts files?

**Review by junlarsen — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (mattgodbolt) [reply package.json]: It was mainly as it complained heartily without me removing this....

File: cypress/tsconfig.json [hunk 2]
```diff
@@ -8,5 +8,7 @@
     /* Code generation */
     "typeRoots": ["../node_modules/@types"],
     "types": ["mocha", "chai", "chai-http"],
+    /* https://github.com/cypress-io/cypress/issues/26203#issuecomment-1571861397 */
```
Comment (mattgodbolt) [inline cypress/tsconfig.json:11]: Thanks!

**Comment by junlarsen** [comment]
> With ts5 are we able to get rid of the .js in imports from ts files?

I think that *might* be possible since we have ts-node but I'm not sure we'd want to do that? ESM module resolution which is the future wants explicit file endings so I think now that we have them, we might as well keep them

**Comment by mattgodbolt** [comment]
Gonna merge! YOLO! :)

**Pull request merged**